### ISSUE TITLE
Payments wait forever to be conducted

### ISSUE THREAD

**Issue by Dominik1999** [raiden-network/raiden#2779]
## Potential Bug / Possible Improvement for the API

Scenario #2778 ran for 5 hours without getting any response code on started payments. Scenario was manually aborted after 5 hours. 
See https://gist.github.com/Dominik1999/ceee2ec4ede7c4f815664610d078998a

I would expect the API to respond with code 408 Requested Timeout.

## Reproduce

Run the following simple [scenario](https://gist.github.com/ulope/9113c794e431e04f460f65e9695cfede).

**Comment by hackaugusto** [comment]
@Dominik1999 the scenario runner logs are not sufficient to figure out what happened. Could you provide the logs from the individual nodes and their reveal timeout? (the timeout of the request depends on the reveal timeout)

**Comment by hackaugusto** [comment]
@Dominik1999 could you get the logs for node `0` too?

**Comment by Dominik1999** [comment]
@hackaugusto 
Here I provide you with everything. It is an easier scenario, but shows the same problem. The Payment from Node:0 to Node:6 never succeeds. 
(Note: In the log-file of Node 0, I only see 5 POST messages, i.e. payments from Node 0 to Node: 1 - Node: 5. The 6th POST message to Node:6 is not in the log file. The 5 other payments were successful in the scenario run. So the problem might be that the scenario player never actually starts the payment, even when he says so)

- scenario: https://gist.github.com/Dominik1999/0abcaec23574d57ee99b5d2a970411c6
- scenario player log file: https://gist.github.com/Dominik1999/477c5229e6a3fb3202d09f26453fef27
- log file node 0: https://gist.github.com/Dominik1999/6b2fad619fe47af39854d7508cdf93d6
- log file node 6 (receiver): https://gist.github.com/Dominik1999/2032cf0708a4d95c5765c4eba0b7386e

I can run the same scenario several times and get this error at different transfers in the scenario. @czepluch faces the same error.

**Comment by czepluch** [comment]
Yes. I just face the problem in an even simpler scenario where only one transfer is being made without any mediators. Ulo tried to run a similar scenario to mine at it works for him. Note that the first time I ran the scenario it succeeded, but not since then even without reusing the same token network.

**Comment by ulope** [comment]
I just discussed this with dominik in a call. The `POST` log messages are misleading since they are only logged **after** the request is finished. 

The log message to look for is logged by the `raiden.api.rest` logger. For example `Initiating payment`.

**Comment by LefterisJP** [comment]
So I had a small look at @czepluch 's 2 logs.

Who is getting stuck? Receiver or sender?

In those logs the sender (`0xe5c68d721324d00e6c90b98b6e6c64250b2bcb9a`) tries multiple times to send payment with identifier `18154601289524244314` to the target `0x2f6df0dc2b8ad73e0e5d75cadd5b0508625e865e`. The token is `0x30e04cb1ccf93585c2228f651369d8d1e9e3d7a1`.

The sender retries many times but the target logs do not show any received transfers. The target only seems to have an active healthcheck with the sender, early in the game:

```
2018-10-16 09:46:29 [debug    ] Healthcheck                    [raiden.network.transport.matrix] current_user=@0x2f6df0dc2b8ad73e0e5d75cadd5b0508625e865e:transport01.raiden.network node=2f6df0dc peer_address= 0xe5c68d721324d00e6c90b98b6e6c64250b2bcb9a
2018-10-16 09:46:29 [debug    ] Changing address presence state [raiden.network.transport.matrix] address=0xe5c68d721324d00e6c90b98b6e6c64250b2bcb9a current_user=@0x2f6df0dc2b8ad73e0e5d75cadd5b0508625e865e:transport01.raiden.network node=2f6df0dc prev_state=None state=<UserPresence.ONLINE: 'online'>
2018-10-16 09:46:29 [debug    ] State change                   [raiden.raiden_service] node=2f6df0dc state_change={"node_address": "0xE5c68D721324D00E6C90B98b6E6C64250B2BcB9a", "network_state": "reachable", "_type": "raiden.transfer.state_change.ActionChangeNodeNetworkState", "_version": 0}
```

And then some logs showing he has acknowledged the channel opening.

```
2018-10-16 09:48:03 [debug    ] Opening channel                [raiden.api.rest] partner_address= 0xE5c68D721324D00E6C90B98b6E6C64250B2BcB9a registry_address=0xbfa863Ac58a3E0A82B58a8e958F2752Bfb573388 settle_timeout=None token_address=0x30e04CB1CCF93585c2228F651369d8d1e9e3D7a1
2018-10-16 09:48:42 [debug    ] Depositing to new channel      [raiden.api.rest] partner_address=0xE5c68D721324D00E6C90B98b6E6C64250B2BcB9a registry_address=0xbfa863Ac58a3E0A82B58a8e958F2752Bfb573388 token_address=0x30e04CB1CCF93585c2228F651369d8d1e9e3D7a1 total_deposit=1000
```
```

**Comment by ulope** [comment]
Yes this is definitely weird. Wild guess: maybe a matrix problem?

**Comment by czepluch** [comment]
@LefterisJP tried to make a transfer to the "receiving node" from another node that has a path while above problem occurs and that transfer also just hangs.

**Comment by ulope** [comment]
More testing yesterday showed that once a transfer between a particular set of nodes has entered this hanging state it doesn't recover even after a node restart and deploying a new token network.

The parameters determining the hanging state seem to be Nodes involved and Eth chain used.

That makes me even more suspicious that the problem may be in the matrix transport, since those three parameters also control room assignment for node-to-node communication.

**Comment by LefterisJP** [comment]
Thank you @ulope for your insight. Is there any way to debug this by using a dedicated matrix server (our testing servers), recreate the problem in its simplest form and watch what happens in Matrix?

**Comment by ulope** [comment]
@LefterisJP You can already observe what's happening. Simply login in (or register an account, if you do that enter no email address in the registration form, it's not supported on our setup) to the matrix server used by the nodes. In this case it was transport01.

In the corresponding room you can see that the initiating node is sending messages but no reply exists from the target.

Unfortunately, as mentioned before I'm not working today, but will look more into this over the weekend.

**Comment by christianbrb** [comment]
@czepluch What is your even simpler scenario you are referring to in this comment? https://github.com/raiden-network/raiden/issues/2779#issuecomment-430174022

**Comment by czepluch** [comment]
@christianbrb As I explain in the comment it was just a simple scenario with one node sending a transfer to another one with no mediators.

**Comment by ulope** [comment]
Here is a simpler scenario that also causes this behavior: https://gist.github.com/ulope/9113c794e431e04f460f65e9695cfede

(Please remember to change the raiden executable path to fit your local environment)

During testing this I also again came across #2838 and discovered a new issue #2879

**Comment by andrevmatos** [comment]
Talking with @konradkonrad and @hackaugusto , I can see two ways for the receiver not be receiving sender messages:
1. A deadlock or other blocking operation happened on the callback handling for a message or event (in `_handle_response` from raiden-libs's GMatrixClient), and this would block the next `/sync` calls and events handling (which is intended, it one event is blocked). We've seen this in the past, and their approach to debug is to install a signal handler to print the stacktrace for all running greenlets to identify these deadlocks. I think that's not what's happening, because this makes the user to also be offline (no /sync -> offline), and if the sender **is** sending the messages, it means the receiver is online.
2. We didn't join or aren't listening for events on this room: that's probably what's happening, and can have a couple of reasons. My main bet is a race where the whitelist of peers we accept invites from (currently populated on `start_health_check`) isn't populated yet during `Transport.start`, when the `initialSync` takes place and the `invite` events sent while we were offline are processed, so these events are ignored, we don't join the room and don't listen to events on it.

I'll make a PR to fix 2., and we can test.

**Comment by christianbrb** [comment]
Added this to the testnet 16 milestone as PR #2889 is needed for PR #2948

**Comment by LefterisJP** [comment]
As discussed in the standup we are not sure if #2948 will fix this. @ulope tried with the WIP PR and it did not seem to have fixed the problem.

**Comment by LefterisJP** [comment]
Note that this issue is only for the messages getting stuck/not being received in the matrix level. There was also an issue of us not properly failing a transfer if the lock expired and waiting on it forever. Using https://github.com/raiden-network/raiden/issues/3094 to track that part and the fix is in https://github.com/raiden-network/raiden/pull/3111
